add feature - generate random problem

yue2388253 opened the pull request:
use "cargo run random" to generate a random problem.

yue2388253 committed:
add feature - generate random problem

use "cargo run random" to generate a random problem.

aylei commented on src/main.rs:
```diff
@@ -16,8 +16,18 @@ fn main() {
+    match id_arg.as_ref() {
+        "random" => {
+            println!("You select random mode.");
+            id = get_random_id();
```
What if the problem is locked or don't have a rust version?

yue2388253 replied:
yes, current implementation has such limitation. 
Do you think we should verify whether the problem of generated id meet these two requirement at first? And regenerate an id that meets these two requirement

yue2388253 replied:
In addition, using random mode to generate problems results in the disorder of lib.rs

aylei replied:
1. I think it is nice to have the ID re-generation, but a proper error message is also okay.
2. It's not a problem, we don't have an order guarantee for `lib.rs`

aylei reviewed (commented):
Rest LGTM.
And this idea is interesting 😜

yue2388253 committed:
Regenerate an id in case the previous generated id is invalide.

Now that using "cargo run" is ok! Then the user could follow the guide of
the system output to do what he/she wants to do (e.g., generate a random
problem or select a particular problem).

yue2388253 commented:
Now that the program can handle the case that the problem has no rust version.

However, I don't know what "the problem is locked" means? Could you please say it in more details?

aylei commented:
> Now that the program can handle the case that the problem has no rust version.
> 
> However, I don't know what "the problem is locked" means? Could you please say it in more details?

I mean some problems are only available for leetcode premium, the subscription plan

yue2388253 committed:
panic when the problem is locked

aylei reviewed (commented):
LGTM, is this PR ready to merge? @yue2388253

yue2388253 commented:
Sorry to late reply. This PR is ready for merge.

BTW, recently I am preparing my paper and thus might not be able to work on this repo.

aylei commented:
@yue2388253 Thanks!
